### DISCUSSION TITLE
Is there away to add bloom to shader lights?

### DISCUSSION THREAD

**Discussion by rosstheblenderanimator** [bnpr/Malt#550] [Q&A] [answered]
https://drive.google.com/file/d/1vGEAIzUebA9-YlMYBQdDXaqCQXNBsUaj/view?usp=drive_link
The reason why I ask was because the spot light won't show up when the environment shadows are on.
![test render 2](https://github.com/bnpr/Malt/assets/146353254/c4408ef2-cf47-4415-a76a-93e98ec9bfb3)
![test render](https://github.com/bnpr/Malt/assets/146353254/f99568f4-1a84-4ffd-9545-8e3f31145a7f)

**Comment by pragma37** [comment]
You can't add bloom to the light shader itself, but you could add a custom output in the Main Pass where you only render those lights and use it as a mask for the bloom.

That's not going to solve your current problem, though, since the lights don't seem to be rendering.
I don't get what you mean with "when the environment shadows are on", a sun light maybe?
If that's the case then it may be a bug.

I see you have linked a google drive file, but it's private and I can't access it.

**Reply by rosstheblenderanimator** [reply]
I updated the link.

**Reply by rosstheblenderanimator** [reply, marked as answer]
I was able to solve the problem with some help from the beer malt bnpr discord channel.
[sample_adjusted.zip](https://github.com/user-attachments/files/16630970/sample_adjusted.zip)
